src/main/scala/io/iohk/ethereum/jsonrpc/PersonalService.scala

```diff
@@ -157,7 +157,12 @@ class PersonalService(
       val maybeCurrentNonce = getCurrentAccount(request.from).map(_.nonce.toBigInt)
       val maybeNextTxNonce = maybeLatestPendingTxNonce.map(_ + 1) orElse maybeCurrentNonce
       val tx = request.toTransaction(maybeNextTxNonce.getOrElse(blockchainConfig.accountStartNonce))
-      val stx = wallet.signTx(tx)
+
+      val stx = if(blockchain.getBlockByNumber(blockchainConfig.eip155BlockNumber).isDefined){
+        wallet.signTx(tx, Some(blockchainConfig.chainId))
+      }else{
+        wallet.signTx(tx, None)
+      }
```

Comment:
Yes, it doesn't address the chain re-org scenario, but after some thought, I don't think that issue is of any practical importance. Worst case, we get some failed TXs in a very short and unlikely timeframe, and only on a test/private network. If you think it's worth the effort and complication in our code then go for it. I doubt it, though.

Nonetheless, I would still prefer using `getBestBlockNumber()` in that condition, just because it's more logical and possibly faster.